qutip/tests/test_states.py

```diff
@@ -66,63 +69,124 @@ def test_implicit_tensor_projection(size, n, m):
     assert implicit == explicit
 
 
-class TestStates:
-    """
-    A test class for the QuTiP functions for generating quantum states
-    """
-
-    def testCoherentState(self):
-        """
-        states: coherent state
-        """
-        N = 10
-        alpha = 0.5
-        c1 = coherent(N, alpha) # displacement method
-        c2 = coherent(7, alpha, offset=3) # analytic method
-        assert_(abs(expect(destroy(N), c1) - alpha) < 1e-10)
-        assert_((c1[3:]-c2).norm() < 1e-7)
-
-
-    def testCoherentDensityMatrix(self):
-        """
-        states: coherent density matrix
-        """
-        N = 10
-
-        rho = coherent_dm(N, 1)
-
-        # make sure rho has trace close to 1.0
-        assert_(abs(rho.tr() - 1.0) < 1e-12)
-
-    def testThermalDensityMatrix(self):
-        """
-        states: thermal density matrix
-        """
-        N = 40
-
-        rho = thermal_dm(N, 1)
-
-        # make sure rho has trace close to 1.0
-        assert_(abs(rho.tr() - 1.0) < 1e-12)
-
-    def testFockDensityMatrix(self):
-        """
-        states: Fock density matrix
-        """
-        N = 10
-        for i in range(N):
-            rho = fock_dm(N, i)
-            # make sure rho has trace close to 1.0
-            assert_(abs(rho.tr() - 1.0) < 1e-12)
-            assert_(rho.data[i, i] == 1.0)
-
-    def testTripletStateNorm(self):
-        """
-        Test the states returned by function triplet_states are normalized.
-        """
-        for triplet in triplet_states():
-            assert_(abs(triplet.norm() - 1.) < 1e-12)
-
-
-if __name__ == "__main__":
-    run_module_suite()
+@pytest.mark.parametrize("base, operator, args, opargs, eigenval", [
+    pytest.param(qutip.basis, qutip.num, (10, 3), (10,), 3,
+                 id="basis"),
+    pytest.param(qutip.basis, qutip.num, (10, 3, 1), (10, 1), 3,
+                 id="basis_offset"),
```

Comment:
it's clearer if you have the id as `basis,offset`, so it doesn't look like we're testing a function `qutip.basis_offset` - by default pytest uses a comma to separate out ids in multi-parametrisation